examples/llm_plus_gnn/rag_feature_store.py

```diff
@@ -0,0 +1,94 @@
+from collections.abc import Iterable, Iterator
+from typing import Any, Dict, Optional, Type, Union
+
+import torch
+from torch import Tensor
+from torch.nn import Module
+from torchmetrics.functional import pairwise_cosine_similarity
+
+from torch_geometric.data import Data, HeteroData
+from torch_geometric.distributed import LocalFeatureStore
+from torch_geometric.nn.nlp import SentenceTransformer
+from torch_geometric.sampler import HeteroSamplerOutput, SamplerOutput
+from torch_geometric.typing import InputEdges, InputNodes
+
+
+# NOTE: Only compatible with Homogeneous graphs for now
+class KNNRAGFeatureStore(LocalFeatureStore):
```

Comment:
u should integrate this into torch_geometric itself, this is a feature, not an example

Reply:
The FeatureStore and Graphstore defined here are meant to be PoCs and cannot scale. I want this example to be a starting point, so that if someone writes to write a FeatureStore/GraphStore interface for a 3rd party service such as Neo4j/Pinecone/etc., they have a starting point. The constraints for the RAG interface are defined by the protocols found in `rag_loader.py`